### DISCUSSION TITLE
mosquitto bridge connection to emqx cloud

### DISCUSSION THREAD

**Discussion by kunj-04-01** [emqx/emqx#14279] [Q&A] [answered]
I am trying bridge connection from mosquitto MQTT explorer to EMQX cloud connection using 8883 port in windows machine. I am facing difficulty connecting to cloud. earlier I had tried to connect using 1883 port and it connected perfectly fine and I received data at EMQX cloud. In the case of 8883 port I tried to connect to cloud but it didn't connect. I downloaded ca certificate from EMQX cloud and gave the path to access certificate in mosquitto.conf file but still it doesn't work. I also created username and password and added to the mosquitto.conf file. which I had created in cloud authentication section. Can anyone check if anything is missing or needs to be changed?

Here's the code that I used in mosquitto.conf file in bridge section.

# Bridge to EMQX Cloud
connection emqx_cloud_bridge
address nae*****.ala.dedicated.aws.emqxcloud.com:8883
# Replace with the username and password provided by EMQX Cloud
username ku****
password ku****
# Topics to bridge
topic # both 0
# Set bridge clean session
cleansession true
# Optional: Configure bridge keepalive (adjust as needed)
keepalive_interval 60
# Optional: Enable start-type automatic connection
start_type automatic
clientid mosquitto_bridge_8883
bridge_cafile C:\Users\emqxcloud-ca.crt
bridge_insecure false                # Validate the server certificate
bridge_tls_version tlsv1.2           # Use TLS 1.2 for secure communication
try_private false                    # Avoid forwarding reserved topics
notifications false                  # Skip $SYS topics

**Comment by id** [comment]
Hi. EMQX Serverless requires SNI. Try to add `--tls-alpn` parameter. See also https://github.com/eclipse-mosquitto/mosquitto/issues/2221.

**Comment by kunj-04-01** [comment, marked as answer]
i looked into the log file and found error related to bridge_tls_version tls1.2
so i comment it and run mosquitto service again by restarting and the bridge got connected to cloud. so the issue got resolved.
